upwind-sensor/default.nix

```diff
@@ -95,41 +95,78 @@ in
       description = "Maximum memory limit for the scanner service (e.g., '500M').";
     };
 
+    cloudProvider = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Explictly set cloud provider; default is auto-detect.";
+    };
+
+    cloudAccountId = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Explicitly set cloud account id; default is auto-detect.";
+    };
+
+    zone = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = "Explicitly set cloud zone/region; default is auto-detect.";
+    };
+
     # TODO: ... add options corresponding to UPWIND_ variables
     # TODO: Configure scanner exclusions
   };
 
   config = mkIf cfg.enable {
-    environment.etc."upwind/sensor.yaml".text = ''
+    # TODO: ... add features like sock-perf, proc-username
+    environment.etc."upwind/agent.yaml".text = ''
       # Upwind Sensor Configuration generated by NixOS
       region: "${cfg.region}"
-      api-host: "https://sensor${if cfg.region == "eu" then ".eu" else ""}.upwind.io" # Derived from script logic
-      auth-endpoint: "https://oauth${if cfg.region == "eu" then ".eu" else ""}.upwind.io/oauth/token" # Derived
+      api-host: "https://agent${if cfg.region == "eu" then ".eu" else ""}.${cfg.domain}"
+      auth-endpoint: "https://oauth${if cfg.region == "eu" then ".eu" else ""}.${cfg.domain}/oauth/token"
       log-level: "${cfg.logLevel}"
-      # ... other config keys based on options and script logic
       install-type: "std"
       platform: "host"
-      # ... add features like sock-perf, proc-username
-    '';
+    ''
+    + (if cfg.cloudProvider != null then "cloud-provider: \"${cfg.cloudProvider}\"\n" else "")
+    + (if cfg.cloudAccountId != null then "cloud-account-id: \"${cfg.cloudAccountId}\"\n" else "")
+    + (if cfg.zone != null then "zone: \"${cfg.zone}\"\n" else "");
 
-    environment.etc."upwind/sensor-hostconfig.yaml".text = ''
+    # TODO: ... other hostconfig specific settings
+    environment.etc."upwind/agent-hostconfig.yaml".text = ''
       # Upwind Hostconfig Configuration generated by NixOS
       region: "${cfg.region}"
-      api-host: "https://sensor${if cfg.region == "eu" then ".eu" else ""}.upwind.io"
-      auth-endpoint: "https://oauth${if cfg.region == "eu" then ".eu" else ""}.upwind.io/oauth/token"
-      log-level: "${cfg.logLevel}" # Assuming same log level option
-      # ... other hostconfig specific settings
+      api-host: "https://agent${if cfg.region == "eu" then ".eu" else ""}.${cfg.domain}"
+      auth-endpoint: "https://oauth${if cfg.region == "eu" then ".eu" else ""}.${cfg.domain}/oauth/token"
+      log-level: "${cfg.logLevel}"
     '';
 
+    # Directory for oauth token
+    systemd.tmpfiles.settings = {
+      "10-upwind-sensor" = {
+        "/var/run/secrets/upwind.io/agent" = {
```

Comment:
This is used?

Reply:
At least as far as it was showing up in the sensor logs as missing when not present. I've not seen it drop any files here however.